test/uk/gov/hmrc/entrydeclarationstore/trafficswitch/TrafficSwitchStateActorSpec.scala

```diff
@@ -0,0 +1,247 @@
+
+          parentProbe.expectMsg(TrafficSwitchActor.SetState(state))
+          parentProbe.expectMsg(TrafficSwitchActor.SetState(state))
+          parentProbe.expectMsg(TrafficSwitchActor.SetState(state))
+
+          stop(stateActor)
+        }
+      }
+
+      "UpdateDatabaseToFlowing is received" must {
+        "update the state in mongo to be not flowing and use notFlowingStateRefreshPeriod" in {
+          val parentProbe = TestProbe()
+
+          MockTrafficSwitchRepo.getTrafficSwitchState returns Future.successful(TrafficSwitchState.Flowing)
+          MockTrafficSwitchRepo.setTrafficSwitchState(TrafficSwitchState.NotFlowing) returns Future.successful(())
+          MockTrafficSwitchRepo.getTrafficSwitchState returns Future.successful(TrafficSwitchState.NotFlowing) anyNumberOfTimes ()
+
+          val stateActor = parentProbe.childActorOf(
+            TrafficSwitchStateActor.props(mockTrafficSwitchRepo, config.copy(notFlowingStateRefreshPeriod = shortTime)))
+
+          parentProbe.expectMsg(TrafficSwitchActor.SetState(TrafficSwitchState.Flowing))
+
+          stateActor ! TrafficSwitchStateActor.UpdateDatabaseToNotFlowing
+          parentProbe.expectMsg(TrafficSwitchActor.AcknowledgeNotFlowing)
+
+          parentProbe.expectMsg(TrafficSwitchActor.SetState(TrafficSwitchState.NotFlowing))
+          parentProbe.expectMsg(TrafficSwitchActor.SetState(TrafficSwitchState.NotFlowing))
+
+          stop(stateActor)
+        }
+
+        // otherwise could reset it to flowing again...
+        "not send further state updates until mongo update has completed" in {
+          val parentProbe = TestProbe()
+
+          MockTrafficSwitchRepo.getTrafficSwitchState returns Future.successful(TrafficSwitchState.Flowing)
+
+          val updatePromise = Promise[Unit]
+          MockTrafficSwitchRepo.setTrafficSwitchState(TrafficSwitchState.NotFlowing) returns updatePromise.future
+          MockTrafficSwitchRepo.getTrafficSwitchState returns Future.successful(TrafficSwitchState.NotFlowing) anyNumberOfTimes ()
+
+          val stateActor = parentProbe.childActorOf(
+            TrafficSwitchStateActor.props(
+              mockTrafficSwitchRepo,
+              config.copy(notFlowingStateRefreshPeriod = shortTime, flowingStateRefreshPeriod = shortTime)))
+
+          stateActor ! TrafficSwitchStateActor.UpdateDatabaseToNotFlowing
+
+          // The parent will ignore any in-flight SetState updates until it receives AcknowledgeNotFlowing
+          // but we must not send any more (even those relating to queries that are currently in progress)
+          // until the DB has been updated
+          parentProbe.fishForMessage() {
+            case _: TrafficSwitchActor.SetState           => false
+            case TrafficSwitchActor.AcknowledgeNotFlowing => true
+          }
+
+          parentProbe.expectNoMessage(100.millis)
+
+          updatePromise.success(())
+
+          // Because of the way that the mocks work, we get the initial flowing
```

Comment:
```suggestion
          // Because of the way that the mocks work, we get the initial TrafficSwitchState.Flowing
```
to be clear we're talking about receiving a specific message